include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2016,7 +2016,7 @@ class binary_reader
             case 'i':
             {
                 std::int8_t number{};
-                if (JSON_HEDLEY_UNLIKELY(!get_number(input_format, number)))
+                if (JSON_HEDLEY_UNLIKELY(!get_number(input_format, number) || number < 0))
```

Comment:
In case of a negative size value - wouldn't it make more sense to create an exception right here with a proper diagnostics like "size values must be positive" or so?

Reply:
done